datafusion/expr/src/utils.rs

```diff
@@ -643,6 +644,35 @@ pub fn expr_as_column_expr(expr: &Expr, plan: &LogicalPlan) -> Result<Expr> {
     }
 }
 
+/// can this data type be used in hash join equal conditions??
+/// If more data types are supported in hash join, add those data types here
+/// to generate join logical plan.
+pub fn can_hash(data_type: &DataType) -> bool {
+    match data_type {
```

Comment:
decimal should be supported here

Reply:
Data types here come from function equal_rows, decimal is not supported in equal_rows.

Reply:
> I think it may help to add a comment here (or in `equal_rows`) mentioning they need to remain in sync

I'll add this comment.